src/TestFramework/Coverage/CodeCoverageDataInterface.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\TestFramework\Coverage;
+
+/**
+ * @internal
+ */
+interface CodeCoverageDataInterface
+{
+    public function hasTests(string $filePath): bool;
+
+    /**
+     * @return CoverageLineData[]
+     */
+    public function getAllTestsForMutation(string $filePath, array $lineRange, bool $isOnFunctionSignature): array;
```

Comment:
what is `$lineRange`?

Reply:
`$lineRange` is the range (of lines) that this node is on. This is especially important for arrays where coverage of them can differ greatly, and we need to figure out exactly all the tests that touch that array.

But it probably isn't that obvious of a name, so i'm open to suggestions. 

As an example

```php
$a = [
    1,
    2,
    3,
];

var_dump(
$a
);
```

In this case the `lineRange` of the node for `$a` would be `[1,2,3,4,5]`. (And due to how we calculate array line ranges, this is the case for every element in that array).

Now for the `var_dump` it would be `[7,8,9]`, but the for the `$a` in there it would be `[8]`

Reply:
Maybe this could be moved to a dedicated value object with that explanation as a comment?